### PR TITLE
feat: Create Search Result Details view

### LINKED ISSUE #128: Create Search Page - Healthcare Professional Details Component
# Requirements
---

- [ ] Create Health Professional Details Component
- [ ] Should be responsive across desktop and mobile view
- [ ] Match visual styling in design prototype
- [ ] Add title, description, type, and languages labels. 
- [ ] Add a SVG from Kei for the photo banner placeholder
- [ ] Add click functionality and selected styling
- [ ] Add a close button

---
# Dependencies

- #128

### PR DESCRIPTION
**Pull request by ermish**
Resolves #128 

30+ commits! this merge contains a lot of work 👍 

# What changed
- [x] Created new generic `Modal` component
- [x] Create new ResultsDetails component
- [x] Created a custom svg for the close button 
- [x] Created svg's for each icon allowing customized colors and stroke widths
- [x] Found a nice doctor svg and customized it to fit our style (doesn't have to be final, just wanted something nice)
- [x] Improved theming consistency and modal styling
- [x] Spent a lot of time fixing bugs
- [x] Cleaned up some css code, file naming, and such 

Known bugs to be fixed later
- [ ] Fixed inconsistent closing action (created #212)
- [ ] modal sizing messes up when map expands and pushes the footer offscreen (created #213)

# Testing instructions

Run the site, check it out in different themes!

### CODE REVIEW

File: app.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-    <div class="h-screen flex flex-col font-sans text-primary-text bg-primary-bg theme-neon">
+    <div class="h-screen flex flex-col font-sans text-primary-text bg-primary-bg theme-violet">
```
Comment (ermish) [inline app.vue:2]: just switching themes so we're not to used to one yet until we have our final colors

File: components/MainContentContainer.vue [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+import SearchResultDetails from './SearchResultDetails.vue';
+
+<template>
+    <div class="flex flex-1 bg-secondary-bg/20 hover:shadow-inner hover:shadow-secondary-bg/90">
+        <!-- <WelcomeSection /> -->
```
Comment (ermish) [inline components/MainContentContainer.vue:6]: this will become the welcomebanner so leaving it in here

File: components/SearchResultDetails.vue [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+<script setup lang="ts">
+import { ref } from 'vue'
+
+const dummyData = {
+    name: 'test person 1',
+    specialty: 'Practitioner',
+    facilityName: 'LifeSavers',
+    spokenLanguages: ['japanese', 'english'],
+    address: '1-2-3 ShinTokyo',
```
Comment (ermish) [inline components/SearchResultDetails.vue:68]: these are temporary types until we have updated ones from the  api schema. 
We do need a finalized one for address in relation to a healthcare provider

File: components/SearchResultsList.vue [hunk 4]
```diff
@@ -15,7 +15,7 @@
     <div :key="index" v-for="(searchResult, index) in searchResults" class="results-list flex flex-col">
         <div class="flex-1 h-24 w-6/8 border-b-2 border-primary/20 p-3
             hover:border-transparent hover:bg-primary/5 transition-all">
-            <SearchResult v-model:name="searchResult.name" v-model:facilityName="searchResult.facilityName"
+            <SearchResultsListItem v-model:name="searchResult.name" v-model:facilityName="searchResult.facilityName"
```
Comment (ermish) [inline components/SearchResultsList.vue:18]: improved clarity of name to distinguish from the detail view of a search result.

**Review by theyokohamalife — APPROVED** [review]
Looking good so far 😎

**Comment by ann-kilzer** [comment]
Looks great!!!

**Pull request merged**